numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2129,6 +2161,18 @@ static PyMethodDef @name@type_methods[] = {
 };
 /**end repeat**/
 
+/**begin repeat
+ * #name = integer,floating#
+ */
+static PyMethodDef @name@type_methods[] = {
+    /* Hook for the round() builtin */
+    {"__round__",
+        (PyCFunction)generic_dunder_round,
+        METH_VARARGS | METH_KEYWORDS, NULL},
+    {NULL, NULL, 0, NULL} /* sentinel */
+};
+/**end repeat**/
+
```

Comment:
Do we want a `np.complexfloating.__round__` that preserves the old behavior with a warning?

Reply:
I tried adding that... I do not know how to suppress the existing warning, and `DeprecationTestCase` doesn't work with two warnings (even though the code is there, it doesn't function).

Reply:
I'd be okay with just removing it, although I can try some more or push incomplete code for help.

Reply:
I would like one that preserves the old behaviour with a warning. Just going to an error directly seems undesirable to me. `DeprecationTestCase` should work with multiple warnings, but you have to tell it that there are multiple warnings.
Unless there is a warning of a different kind, in which case you may have to ignore that using pytest (or possibly `suppress_warnings` just because that is how `DeprecationTestCase` was originally written).